apps/terminal/api/component/terminal.py

```diff
@@ -89,3 +91,17 @@ def create(self, request, *args, **kwargs):
             return Response(data=data, status=status.HTTP_400_BAD_REQUEST)
         return super().create(request, *args, **kwargs)
 
+
+class EncryptedTerminalConfig(generics.CreateAPIView):
+    serializer_class = serializers.EncryptedConfigSerializer
+    permission_classes = [IsServiceAccount]
+    http_method_names = ['post']
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        encrypt_key = serializer.validated_data['secret_encrypt_key']
+        encrypted_value = serializer.validated_data['encrypted_value']
+        config_crypto = ConfigCrypto(encrypt_key)
+        value = config_crypto.decrypt(encrypted_value)
+        return Response(data={'value': value}, status=200)
```

Comment:
The main difference compared to the previous version of the TerminalConfig serializer is that it requires both `secret_encrypt_key` and `encrypted_value`. Also, `EncryptedTerminalConfig` class has been moved out of its own namespace from `.api`, implying this new class doesn't need to depend on other classes in Django's app. The change will make this view more clean for development purposes and potentially improve code readability.

However, before considering these changes, we should ensure there are no regressions due to altering the structure without properly documenting them. Additionally, if security concerns arise after deployment, consider reevaluating whether an approach with secret encryption keys might be more suitable instead.